### PR TITLE
PR for polishing

### PR DESCRIPTION
**Pull request by t-b**

### CODE REVIEW

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 1]
```diff
@@ -100,6 +103,7 @@ NDArray* NDArrayPool::alloc(int ndims, size_t *dims, NDDataType_t dataType, size
     }
     /* Erase the attributes if that global flag is set */
     if (eraseNDAttributes) pArray->pAttributeList->clear();
+    // calcs totalBytes
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp]: the surrounding code uses c style comments we should prefer that here too.

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 2]
```diff
@@ -109,53 +113,103 @@ NDArray* NDArrayPool::alloc(int ndims, size_t *dims, NDDataType_t dataType, size
...
   }
 
+  if (freeArray && pArray) {
+	  delete pArray;
+
+	  pArray = freeArray;
+	  NDArray* nextArray = (NDArray *)ellNext(&freeArray->node);
+	  size_t diffSize;
+	  if (pArray->dataSize > dataSize)
+	  {
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp]: the rest of the code seems to prefer `{` on the same line as if/while etc.

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 3]
```diff
@@ -225,13 +279,7 @@ int NDArrayPool::reserve(NDArray *pArray)
          driverName, functionName, pArray->pNDArrayPool, this);
     return(ND_ERROR);
   }
-  //printf("NDArrayPool::reserve pArray=%p, count=%d\n", pArray, pArray->referenceCount);
   epicsMutexLock(listLock_);
-  // If the reference count is less than 1 then something is wrong, this NDArray has been released.
-  if (pArray->referenceCount < 1) {
-    cantProceed("%s:reserve ERROR, reference count = %d, should be >= 1, pArray=%p\n",
-           driverName, pArray->referenceCount, pArray);
-  }
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp]: what is the rationale for removing this code?

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 4]
```diff
@@ -255,7 +303,6 @@ int NDArrayPool::release(NDArray *pArray)
            driverName, functionName, pArray->pNDArrayPool, this);
     return(ND_ERROR);
   }
-  //printf("NDArrayPool::release pArray=%p, count=%d\n", pArray, pArray->referenceCount);
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp]: yes it is ugly but just leave it. It's not our codebase ;)

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 5]
```diff
@@ -264,7 +311,7 @@ int NDArrayPool::release(NDArray *pArray)
     numFree_++;
   }
   if (pArray->referenceCount < 0) {
-    cantProceed("%s:release ERROR, reference count < 0 pArray=%p\n",
+    printf("%s:release ERROR, reference count < 0 pArray=%p\n",
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp]: why the change from cantProceed to printf?

File: ADApp/ADSrc/NDArrayPool.cpp [hunk 6]
```diff
@@ -644,7 +691,18 @@ int NDArrayPool::report(FILE *fp, int details)
         (long)memorySize_, (long)maxMemory_);
   fprintf(fp, "  numFree=%d\n",
          numFree_);
-      
+  if(details > 0)
+  {
+    NDArray* freeArray = (NDArray *)ellFirst(&freeList_);
+    unsigned i = 0;
```
Comment (t-b) [inline ADApp/ADSrc/NDArrayPool.cpp:700]: Looks good but in a separate commit.

**Comment by MichaelHuth** [comment]
Ok I have changes/updated the commits.

**Comment by t-b** [comment]
@MichaelHuth Thanks. Please bring it upstream.
